### DISCUSSION TITLE
Authentication response is correct but the resolved result of "authWithPassword" has empty values

### DISCUSSION THREAD

**Discussion by ilearnio** [pocketbase/pocketbase#2674] [Q&A]
I'm trying to implement authentication using by following the official guide here https://pocketbase.io/docs/authentication/, so my code looks like so:

```ts
const authData = await pb.collection('users').authWithPassword(email, password)
console.log('authData:', authData)

console.log('pb.authStore.isValid:', pb.authStore.isValid)
console.log('pb.authStore.token:', pb.authStore.token)
console.log('pb.authStore.model?.id:', pb.authStore.model?.id)
  ```
  
  However when signing in, I see this in the console:
  
  
![image](https://github.com/pocketbase/pocketbase/assets/14331216/5eb4a67e-8e12-4232-95ce-f5673a1fc312)

even though the response is correct:

![image](https://github.com/pocketbase/pocketbase/assets/14331216/04e3ccd3-b8e4-499f-9791-aa451810e57f)

Also the user record in the local storage is saved with empty values:

![image](https://github.com/pocketbase/pocketbase/assets/14331216/60aee492-b48a-4810-97e8-4a094ae21874)

I may be wrong but It looks like a bug on Pocketbase end

**Comment by ganigeorgiev** [comment]
I'm not able to reproduce it locally with JS SDK v0.15.2.

Is the above code executed in a SSR environment? Could you provide a more complete code sample?

To rule out PocketBase issue, if you use the prebuilt executable you can create a `pb_public/index.html` with the following content and check the console at `http://127.0.0.1:8090`:

```html
<!DOCTYPE html />
<html>
    <head>
        <meta charset="utf-8" />
        <title>Auth test</title>
    </head>
    <body>
        <script type="module">
            import PocketBase from "https://cdn.jsdelivr.net/gh/pocketbase/js-sdk@v0.15.2/dist/pocketbase.es.mjs"

            const pb = new PocketBase("http://127.0.0.1:8090");

            const data = await pb.collection("users").authWithPassword("test@example.com", "1234567890")
            
            console.log(data);
        </script>
    </body>
</html>
```

**Comment by ilearnio** [comment]
Sorry, it was my bad. I had this "afterSend" hook, which looked like so:
```ts
pb.afterSend = (response, data) => {
  const userStore = useUserStore()
  // 401 Unauthorized; 419 Session expired
  if ([401, 419].includes(response.status)) {
    userStore.logoutUser()
    return
  }
}
```
 which didn't return any value (didn't know it should), so after I added `return data` at the end of it, the issue with login was fixed.
 
 Thanks for the quick response!

**Reply by ganigeorgiev** [reply]
Yes, the `afterSend` hook must return the data that will be resolved. Currently it is documented only in the [jsdoc of the method](https://github.com/pocketbase/js-sdk/blob/04b699fd4218f9579655f31350807fa7f7073286/src/Client.ts#L58-L78). I'll consider sometime in the future to generate a html docs from the code comments for better visibility.
